### PR TITLE
Data tests refactor 

### PR DESCRIPTION
**Pull request by MXPOL**
# Reason for This PR

The new feature query validator in `data.js` broke some test, this PR refactors a  little the test to pass the query validator.

### CODE REVIEW

File: packages/velo-external-db-commons/lib/data_commons.js [hunk 1]
```diff
@@ -57,6 +57,19 @@ const AdapterOperators = {
     urlized: 'urlized'
 }
 
+const WixOperators = {
```
Comment (MXPOL) [inline packages/velo-external-db-commons/lib/data_commons.js:60]: We generate a random Wix operator in `gen.js`, so for the sake of order created a new enum, the random-wix-operator function takes a random value from this enum.

File: packages/velo-external-db-commons/lib/data_commons.js [hunk 2]
```diff
@@ -65,10 +78,44 @@ const AdapterFunctions = {
     count: 'count'
 }
 
+const wixOperatorToAdapterOperator = operator => {
```
Comment (MXPOL) [inline packages/velo-external-db-commons/lib/data_commons.js:81]: Extract this function from `filter_transformer.js` because I need it also in tests and I didn't want to duplicate myself

File: packages/velo-external-db-core/lib/service/data.spec.js [hunk 3]
```diff
@@ -168,6 +171,8 @@ describe('Data Service', () => {
         ctx.itemId = chance.guid()
         ctx.itemIds = Array.from({ length: 10 }, () => chance.guid())
         ctx.total = chance.natural({ min: 2, max: 20 })
+        ctx.field = gen.randomDbField()
```
Comment (MXPOL) [inline packages/velo-external-db-core/lib/service/data.spec.js:174]: Generate random column, this column will be the only column in the schema that will be return be the mocked 
`CacheableSchemaInformation`

File: packages/velo-external-db-core/lib/service/data.spec.js [hunk 4]
```diff
@@ -168,6 +171,8 @@ describe('Data Service', () => {
...
+        ctx.filterField = gen.randomFilterBaseOnField(ctx.field.field, ctx.field.type)
```
Comment (MXPOL) [inline packages/velo-external-db-core/lib/service/data.spec.js:175]: Because of the  query validator, we need to generate a valid filter (a column that exists in schema, and an operation that can be preformed of the column)

File: packages/velo-external-db-core/test/drivers/data_provider_test_support.js [hunk 5]
```diff
@@ -12,7 +12,7 @@ const dataProvider = {
 }
 
 const givenListResult = (entities, forCollectionName, filter, sort, skip, andLimit) =>
-    when(dataProvider.find).calledWith(forCollectionName, filter, sort, skip, andLimit)
+    when(dataProvider.find).calledWith(forCollectionName, expect.anything(), sort, skip, andLimit)
```
Comment (MXPOL) [inline packages/velo-external-db-core/test/drivers/data_provider_test_support.js:15]: Here we stub the data-provider find function that `data.js` uses, because of the new filter I need the filter that the `filterTransformer.transform` creates and I don't have it, because of this I changed it to `expect.anything()` 
@noam-almog  Is it terrible?

Comment (noam-almog) [reply packages/velo-external-db-core/test/drivers/data_provider_test_support.js:15]: i don't understand why can't you know what the input is

**Comment by MXPOL** [comment]
Closing the PR, we need a new PR with a different approach: 
 
In `data.js` tests, we don't need to supply the right values for the validate function, we just need to ensure that it's not throwing any error, we can do it by mock the validate function and set the response that the function will return, we are testing `data.js` functions and not the function that it uses ... 

Another thing I think it will be better that `queryValidator.validateFilter` will not throw errors, instead it will just return the fields that doesn't exist or operations that cannot be done on fields, and `data.js` will throw an error depends on the result. (Like I did in `schema.js`)
